src/tests/java/picard/vcf/TestFilterVcf.java

```diff
@@ -41,7 +43,33 @@
 public class TestFilterVcf {
     private final File INPUT = new File("testdata/picard/vcf/filter/testFiltering.vcf");
     private final File BAD_INPUT = new File("testdata/picard/vcf/filter/testFilteringNoSeqDictionary.vcf");
-
+    
+    /* write content of javascript in the returned file */
+	private File quickJavascriptFilter(String content) throws Exception {
+		final File out = File.createTempFile("jsfilter", ".js");
+		out.deleteOnExit();
+		PrintWriter pw = new PrintWriter(out);
+		pw.println(content);
+		pw.flush();
+		pw.close();
+		return out;
+	}
+
+	@Test
+	public void testJavaScript() throws Exception {
+		final File out = File.createTempFile("filterVcfTestJS.", ".vcf");
+		out.deleteOnExit();
+		final FilterVcf filterer = new FilterVcf();
+		filterer.INPUT = INPUT;
+		filterer.OUTPUT = out;
+		filterer.JAVASCRIPT_FILE = quickJavascriptFilter("variant.getStart()%5 != 0");
+
+		final int retval = filterer.doWork();
+		if (retval != 0) {
+			throw new PicardException("Return value non-zero: " + retval);
+		}
+	}
+    
```

Comment:
please test the samfilter as well. also, it would be nice to test that it actually worked...as in, did the thing you wanted, not just completed successfully.